continuous-integration/readme.md

```diff
@@ -1,3 +1,233 @@
+- [ ] **Infrastructure Documentation**
+  - The deployment and cloud resource template topology should be documented and well understood within the README of the IAC git repo.
+  - Local environment and CI workflow setup steps should be documented.
+
+## Integration Validation
+
+An effective way to identify bugs in your build at a rapid pace is to invest early into a reliable suite of automated tests that validate the baseline functionality of the system:
+
+- [ ] **End to end integration tests**
+  - Include tests in your pipeline to validate the build candidate conforms to automated business functionality assertions. Any bugs or broken code should be reported in the test results including the failed test and relevant stack trace. All tests should be invoked through a single command.
+  - Keep the build fast. Consider automated test runtime when deciding to pull in dependencies like databases, external services and mock data loading into your test harness. Slow builds often become a bottleneck for dev team's when parallel builds on a CI server aren't an option. Consider adding max timeout limits for lengthy validations to fail fast and maintain high velocity across the team.
+
+- [ ] **Avoid checking in broken builds**
+  - Automated build checks, tests, lint runs, etc should be validated locally before committing your changes to the scm repo. [Test Driven Development](https://martinfowler.com/bliki/TestDrivenDevelopment.html) is a practice dev crews should consider to help identify bugs and failures as early as possible within the development lifecycle.
+- [ ] **Reporting build failures**
+  - If the build step happens to fail then the build pipeline run status should be reported as failed including relevant logs and stacktraces.
+- [ ] **Test Automation Data Dependencies**
+  - Any mocked dataset(s) used for unit and end-to-end integration tests should be checked into the mainline repository. Minimize any external data dependencies with your build process.
+
+## Git Driven Workflow
+
+- [ ] **Build on commit**
+  - Every commit to the baseline repository should trigger the CI pipeline to create a new build candidate.
+  - Build artifact(s) are built, packaged, validated and deployed continuously into a non-production environment per commit. Each commit against the repository results into a CI run which checks out the sources onto the integration machine, initiates a build, and notifies the committer of the result of the build.
+
+- [ ] **Master branch consistency with production**
+  - Merges into the master branch trigger releases into the production environment(s). This way the master branch becomes a dependable baseline for the code running in production.
+
+- [ ] **Avoid commenting out failing tests**
+  - Avoid commenting out tests in the mainline branch. By commenting out tests, we get an incorrect indication of the status of the build.
+
+- [ ] **Branch policy enforcement**
```

Comment:
add this link https://help.github.com/en/github/administering-a-repository/configuring-protected-branches